workitem/link/link_repository_blackbox_test.go

```diff
@@ -187,6 +190,216 @@ func (s *linkRepoBlackBoxTest) TestCreate() {
+					// then
+					if errorExpected[1] {
+						require.Error(t, err)
+					} else {
+						require.NoError(t, err)
+					}
+				})
+				//  III:
+				//
+				//   C
+				//    *
+				//     A
+				//      \
+				//       C
+				//        \
+				//         B
+				t.Run("Scenario III", func(t *testing.T) {
+					// given
+					fxt := tf.NewTestFixture(t, s.DB,
+						tf.WorkItems(3, tf.SetWorkItemTitles("A", "C", "B")),
+						tf.WorkItemLinkTypes(1, tf.SetTopologies(topo)),
+						tf.WorkItemLinks(2, func(fxt *tf.TestFixture, idx int) error {
+							l := fxt.WorkItemLinks[idx]
+							switch idx {
+							case 0:
+								l.SourceID = fxt.WorkItemByTitle("A").ID
+								l.TargetID = fxt.WorkItemByTitle("C").ID
+							case 1:
+								l.SourceID = fxt.WorkItemByTitle("C").ID
+								l.TargetID = fxt.WorkItemByTitle("B").ID
+							}
+							return nil
+						}),
+					)
+					// when
+					_, err := s.workitemLinkRepo.Create(s.Ctx, fxt.WorkItemByTitle("C").ID, fxt.WorkItemByTitle("A").ID, fxt.WorkItemLinkTypes[0].ID, fxt.Identities[0].ID)
+					// then
+					if errorExpected[2] {
+						require.Error(t, err)
+					} else {
+						require.NoError(t, err)
+					}
+				})
+				//  IV:
+				//
+				//   C
+				//    *
+				//     A
+				//      \
+				//       B
+				t.Run("Scenario IV", func(t *testing.T) {
+					// given
+					fxt := tf.NewTestFixture(t, s.DB,
+						tf.WorkItems(3, tf.SetWorkItemTitles("A", "B", "C")),
+						tf.WorkItemLinkTypes(1, tf.SetTopologies(topo)),
+						tf.WorkItemLinks(1, func(fxt *tf.TestFixture, idx int) error {
+							l := fxt.WorkItemLinks[idx]
+							switch idx {
+							case 0:
+								l.SourceID = fxt.WorkItemByTitle("A").ID
+								l.TargetID = fxt.WorkItemByTitle("B").ID
+							}
+							return nil
+						}),
+					)
+					// when
+					_, err := s.workitemLinkRepo.Create(s.Ctx, fxt.WorkItemByTitle("C").ID, fxt.WorkItemByTitle("A").ID, fxt.WorkItemLinkTypes[0].ID, fxt.Identities[0].ID)
+					// then
+					if errorExpected[3] {
+						require.Error(t, err)
+					} else {
+						require.NoError(t, err)
+					}
+				})
+				//  V:
+				//
+				//   A
+				//    \
+				//     B
+				//      *
+				//       C
+				t.Run("Scenario V", func(t *testing.T) {
+					// given
+					fxt := tf.NewTestFixture(t, s.DB,
+						tf.WorkItems(3, tf.SetWorkItemTitles("A", "B", "C")),
+						tf.WorkItemLinkTypes(1, tf.SetTopologies(topo)),
+						tf.WorkItemLinks(1, func(fxt *tf.TestFixture, idx int) error {
+							l := fxt.WorkItemLinks[idx]
+							switch idx {
+							case 0:
+								l.SourceID = fxt.WorkItemByTitle("A").ID
+								l.TargetID = fxt.WorkItemByTitle("B").ID
+							}
+							return nil
+						}),
+					)
+					// when
+					_, err := s.workitemLinkRepo.Create(s.Ctx, fxt.WorkItemByTitle("B").ID, fxt.WorkItemByTitle("C").ID, fxt.WorkItemLinkTypes[0].ID, fxt.Identities[0].ID)
```

Comment:
SourceID should be C
TargetID should be B

Reply:
@DhritiShikhar no, C is the child of B. Therefore C is the target.